types/markdownlint-rule-helpers/index.d.ts

```diff
@@ -0,0 +1,380 @@
+// Type definitions for markdownlint-rule-helpers 0.21
+// Project: https://github.com/DavidAnson/markdownlint
+// Definitions by: Josh Goldberg <https://github.com/JoshuaKGoldberg>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+/// <reference types="node" />
+
+import * as MarkdownIt from 'markdown-it';
+
+export type RuleParams = any;
```

Comment:
I couldn't figure out from the code what this is supposed to be 🤔 

* https://github.com/DavidAnson/markdownlint/blob/7f14d9b5a861b31938f72a60ce03d50b269ae251/helpers/helpers.js#L328
* https://github.com/DavidAnson/markdownlint/blob/7f14d9b5a861b31938f72a60ce03d50b269ae251/helpers/helpers.js#L578
* https://github.com/DavidAnson/markdownlint/blob/7f14d9b5a861b31938f72a60ce03d50b269ae251/helpers/helpers.js#L328

cc @DavidAnson

Reply:
The `RuleParams` type is defined here: https://github.com/DavidAnson/markdownlint/blob/7f14d9b5a861b31938f72a60ce03d50b269ae251/lib/markdownlint.d.ts#L111-L135

Reply:
Thanks! Pulled in all the types I could find from there.

Reply:
Thanks! Pulled in all the types I could find from there.

I hadn't seen the `.d.ts` file in the `markdownlint` repo. Would you rather I moved these newly added types to the repo instead?

Reply:
Since markdownlint bundles its own types, maybe it's better to commit markdownlint-rule-helpers types to markdownlint's repository too.
And the markdownlint types in this repository can be removed later.

Reply:
The `markdownlint-rule-helpers` package is unsupported and gets breaking changes in most releases and so I deliberately do not provide documentation or typings for it.

Reply:
I thought I sent a PR to this repo at one point to remove the old/outdated types for `markdownlint`, but perhaps I am remembering wrong or they got added back. They sound redundant given the types I maintain in the project itself.

Reply:
@plantain-00, here is the merged PR from a couple of years ago where I did that. I don't know what this project's policy on redundancy is.

https://github.com/DefinitelyTyped/DefinitelyTyped/pull/46826

Reply:
Sorry, I mixed markdownlint and markdown-it.

Reply:
https://github.com/microsoft/DefinitelyTyped-tools/pull/707 👍